src/main/java/net/logstash/logback/composite/CompositeJsonFormatter.java

```diff
@@ -118,6 +118,56 @@ public boolean isStarted() {
         return started;
     }
 
+    /**
+     * Create a reusable {@link JsonFormatter} bound to the given {@link OutputStream}.
+     * 
+     * @param outputStream the output stream used by the {@link JsonFormatter}
+     * @return {@link JsonFormatter} writing JSON content in the output stream
+     * @throws IOException thrown when unable to write in the output stream or when Jackson fails to produce JSON content
+     */
+    public JsonFormatter createJsonFormatter(OutputStream outputStream) throws IOException {
+        if (!isStarted()) {
+            throw new IllegalStateException("Formatter is not started");
+        }
+        
+        JsonGenerator generator = createGenerator(outputStream);
+        return new JsonFormatter(generator);
+    }
+
+    /**
+     * Create a reusable {@link JsonFormatter} bound to the given {@link Writer}.
+     * 
+     * @param writer the {@link Writer} used by the {@link JsonFormatter}
+     * @return {@link JsonFormatter} writing JSON content in the writer
+     * @throws IOException thrown when unable to write in the writer or when Jackson fails to produce JSON content
+     */
+    public JsonFormatter createJsonFormatter(Writer writer) throws IOException {
+        if (!isStarted()) {
+            throw new IllegalStateException("Formatter is not started");
+        }
+        
+        JsonGenerator generator = createGenerator(writer);
+        return new JsonFormatter(generator);
+    }
+    
+    
+    public class JsonFormatter {
+        private final JsonGenerator generator;
+        
+        public JsonFormatter(JsonGenerator generator) {
+            this.generator = generator;
+        }
+        
+        public void writeEvent(Event event) throws IOException {
+            writeEventToGenerator(generator, event);
+        }
+        
+        public void dispose() throws IOException {
+            this.generator.close();
+        }
+    }
```

Comment:
_nit_ Perhaps rename to `close()` and make the class implement `java.io.Closeable`.